Add driver for Vishay RGBIR sensor VEML6046

it-klinger opened the pull request:
- add driver for Vishay VEML6046 RGBIR color sensor
- add new compatible "vishay,veml6046"
- support fetch and get sensor subsystem operations
- triggered mode and interrupts are not yet supported
- add attribution test application which tests all attribute combinations

it-klinger committed:
drivers: sensor: add Vishay veml6046 RGBIR color sensor

- add driver for Vishay VEML6046 RGBIR color sensor
- add new compatible "vishay,veml6046"
- support fetch and get sensor subsystem operations
- triggered mode and interrupts are not yet supported

Signed-off-by: Andreas Klinger <ak@it-klinger.de>

it-klinger committed:
samples: sensor: veml6046: add attribution test application

- Test all attribute combinations of Vishay RGBIR color sensor VEML6046.
- Print OVERFLOW in case of saturation of sensor.
- This small program is intended to be helping when finding appropriate
  attributes for an application of the sensor.

Signed-off-by: Andreas Klinger <ak@it-klinger.de>

it-klinger commented on drivers/sensor/vishay/veml6046/veml6046.c:
```diff
@@ -0,0 +1,631 @@
+	uint32_t green_lux;
+	uint32_t blue_lux;
+	uint32_t ir_lux;
+	uint32_t int_flags;
```
Unused. I'll remove.

it-klinger commented:
Hi, 
can anybody review the patch, please?
I think i changed everything according to the review or did i miss something?

kartben commented:
> Hi, 
> can anybody review the patch, please?
> I think i changed everything according to the review or did i miss something?

It will be possible as soon as you restore https://github.com/it-klinger/zephyr/tree/add-veml6046 to commit d7051b9 which is what it was when the PR was closed. Hope this helps!

it-klinger commented:
Hi Benjamin,
i restored to d7051b9 and also did a small change and force commited, but the PR is not opening again.

kartben commented:
> Hi Benjamin,
> i restored to d7051b9 and also did a small change and force commited, but the PR is not opening again.

It's only when the branch is exactly at that commit that reopening will be possible. The follow-up commit will have to be done afterwards

it-klinger commented:
Now it's at d705 but i'm only able to open a PR with a new number (#94xxx), but the closed one is not opened.

kartben commented:
@it-klinger re-opened now :) thanks for your perseverance

JarmouniA reviewed (commented):
Squash the "Update...." commits.

josuah reviewed (commented):
Not immediately approving just to give time for an extra verification. I am not sure if I misunderstood some detail or if there's a typo.

Some extra non-blocking minor hints about the doc and other things too. :)

I did not test with hardware.

Thank you!

kartben reviewed (changes requested):
see comments
also, it feels like you want to come up with some common code/headers with VEML6031?

it-klinger committed:
drivers: sensor: remove redundancies in veml6031 and veml6046

- create common header file veml60xx-common.h for sensors VEML6031 and
  VEML6046.

Signed-off-by: Andreas Klinger <ak@it-klinger.de>

it-klinger commented on include/zephyr/drivers/sensor/veml6046.h:
```diff
@@ -0,0 +1,175 @@
+/**
+ * @brief Gain selections in enum veml6046_gain.
+ */
+#define VEML6046_GAIN_COUNT 4
```
@kartben Thanks for the review and the example on #96270 . It helped a lot.

josuah reviewed (dismissed):
LGTM, thank you for the modifications!

kartben commented on drivers/sensor/vishay/veml6046/veml6046.c:
```diff
@@ -0,0 +1,594 @@
+			return -ENOTSUP;
+		}
+	}
+	return 0;
```
seems like a valid comment to me, unless I am missing something

josuah reviewed (dismissed):
Thank you for the successive modifications and improvements!

it-klinger commented:
@MaureenHelm @kartben @josuah 

Rebasing done.
